### PR TITLE
🧹 Add "Build & Test" Workflow

### PR DESCRIPTION
**Pull request by pje**
### Motivation

If you haven't noticed, we've got a lot of [issues](https://github.com/actions/labeler/issues) piling up! I'm going to be going through the open issues/PRs in the near future, and in order to confidently make changes to this codebase, we need—at the minimum—an automated test suite.  This PR just gets that ball rolling!

### Next Steps

Actually write some meaningful tests.

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+on:
+  pull_request:
+    paths-ignore:
+      - '**.md'
+  push:
+    branches:
+      - main
+      - releases/*
+      - test-build
```
Comment (TingluoHuang) [inline .github/workflows/build_test.yml:11]: what is `test-build` ?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "labeler",
-  "version": "3.0.0",
+  "version": "3.0.1",
```
Comment (TingluoHuang) [inline package.json:3]: do we need to bump this in this PR?

Comment (pje) [reply package.json:3]: I can bump it in a subsequent commit if that's our workflow—as part of my cleanup in this repo I'll add guidelines on our release process too, since there aren't any right now.

Comment (TingluoHuang) [reply package.json:3]: I would suggest not bump it in this PR, I think we normally bump it when we are ready to release a newer version.

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,6415 @@
 {
   "name": "labeler",
-  "version": "3.0.0",
-  "lockfileVersion": 1,
+  "version": "3.0.1",
```
Comment (TingluoHuang) [inline package-lock.json:3]: undo this change as well?

**Review by TingluoHuang — APPROVED** [review]

**Comment by pje** [comment]
Thanks @TingluoHuang! Will squash 'n' merge.

**Comment by pje** [comment]
https://github.com/github/c2c-actions-runtime/issues/1273
